spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -146,6 +153,31 @@ data class SpdxCompoundExpression(
 ) : SpdxExpression() {
     override fun decompose() = left.decompose() + right.decompose()
 
+    override fun disjunctiveNormalForm(): SpdxExpression {
+        val leftDnf = left.disjunctiveNormalForm()
+        val rightDnf = right.disjunctiveNormalForm()
+
+        return when (operator) {
+            SpdxOperator.OR -> SpdxCompoundExpression(leftDnf, SpdxOperator.OR, rightDnf)
+            SpdxOperator.AND -> {
+                when {
+                    leftDnf is SpdxCompoundExpression && leftDnf.operator == SpdxOperator.OR &&
```

Comment:
No, I just made it up. Existing algorithms for boolean expressions are more complex because they have to deal with the NOT operator which does not exist in SPDX.